### PR TITLE
Update trajectory prediction

### PR DESCRIPTION
**Pull request by patrick-gu**
- Update drag calculation to use final CFD simulation values
- Refactor trajectory prediction math into its own thing so it can be integrated with OR

Todo:
- Update the mass of the rocket
- Pull controller logic out into its own thing as well so it can be tested

### CODE REVIEW

**Comment by celery6** [comment]
while you happen to be moving stuff around.. any chance you have time to move our random dependency/submodules into a subdir? This would include `Calculations`, `Fusion` `canlib` `printf` `vnproglib` and ideally `FATFS` but I think `FATFS` might (?) be managed by the code-generator so it's stuck there.

Also the `millis.c` and `millis.h` in `Tasks` could be its own lib the way `printf` is. 

Also the `ICM-20948` files in `Tasks` could be somewhere else too - these are more like hardwarre drivers, not Tasks.

None of these changes should affect any work in progress currently so it theoretically wouldn't cause merge conflicts for any branch if we do it now

File: STM32Cube/Tasks/trajectory.h [hunk 1]
```diff
@@ -15,27 +15,13 @@ extern xQueueHandle altQueue;
...
-typedef struct ForceStruct{
-    float Fy; //N
-    float Fx; //N
-} Forces;
-
-//Struct for recieving altitude data, could replace with import
-typedef struct AltStruct{
-    int32_t alt; //altitude in m
-    float time; //time in ms
+// Struct for recieving altitude data, could replace with import
```
Comment (celery6) [inline STM32Cube/Tasks/trajectory.h:21]: could repl w/ what import? no clue what this is referring to but this struct shld prob be in traj lib with all the other data structs

File: STM32Cube/Calculations/trajectory_lib.c [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+    if (extension < 0) {
+        extension = 0;
+    } else if (extension > 1) {
+        extension = 1;
+    }
+
+    float x = (velocity - 273.9f) / 148.7f;
+    float y = (altitude - 5000.0f) / 3172.0f;
+
+    Cubic2VariablePolynomial coeffs[11] = {
```
[the remark this answers is not in the document]
Comment (patrick-gu) [reply STM32Cube/Calculations/trajectory_lib.c:60]: They come from Matlab, and I think if I made a comment it's still not exactly human readable

File: STM32Cube/Calculations/trajectory_lib.c [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+        {340.9146f, 367.1520f, -114.8622f, 101.0439f, -123.1214f, 18.4047f,
+         -0.1879f, -31.8071f, 15.4422f, -1.1456f}};
+
+    int index = (int)(extension * 10.0f);
+    if (extension * 10.0f == index) {
+        return evaluate_cubic_2_variable(&coeffs[index], x, y);
+    }
+    float first = evaluate_cubic_2_variable(&coeffs[index], x, y);
+    float second = evaluate_cubic_2_variable(&coeffs[index + 1], x, y);
+    float diff = extension - ((float)((int)extension));
```
[the remark this answers is not in the document]
Comment (patrick-gu) [reply STM32Cube/Calculations/trajectory_lib.c:90]: Mistake, I meant to take the fractional part of ext * 10

File: STM32Cube/Tasks/trajectory.h (hunk 1 above)
Comment (patrick-gu) [reply STM32Cube/Tasks/trajectory.h:21]: This was already there, don't know what it means

File: STM32Cube/Calculations/trajectory_lib.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (patrick-gu) [reply STM32Cube/Calculations/trajectory_lib.c:90]: Linear interpolation can be deleted actually since we're going to have fixed extension

**Comment by patrick-gu** [comment]
Can I descope other moving things around because I honestly don't really know how to do that

**Comment by Joe-Joe-Joe-Joe** [comment]
yes

**Review by Joe-Joe-Joe-Joe — APPROVED** [review]
behaves as expected. LGTM

**Pull request merged**